cluster.go

```diff
@@ -112,7 +112,7 @@ func (c *clusterClient) _refresh() (err error) {
 		return err
 	}
 
-	groups := parseSlots(reply)
+	groups := parseSlots(reply, c.opt.InitAddress[0])
```

Comment:
Always using the first address is probably not correct. According to the document, the address should be the one where the reply was sent from.

Reply:
I think we might need to change the `results` from being
```go
results := make(chan RedisResult, len(c.conns))
```
to being
```go
results := make(chan struct{reply RedisResult; conn conn}, len(c.conns))
```
So that we can keep track of reply and conn pairs and use `conn.Addr()` later to get the reply address.